packages/web3-core-method/src/index.js

```diff
@@ -526,16 +553,34 @@ var getWallet = function(from, accounts) {
 
 Method.prototype.buildCall = function() {
     var method = this,
-        isSendTx = (method.call === 'eth_sendTransaction' || method.call === 'eth_sendRawTransaction'); // || method.call === 'personal_sendTransaction'
+        isSendTx = (method.call === 'eth_sendTransaction' || method.call === 'eth_sendRawTransaction'), // || method.call === 'personal_sendTransaction'
+        isCall = (method.call === 'eth_call');
 
     // actual send function
     var send = function() {
         var defer = promiEvent(!isSendTx),
             payload = method.toPayload(Array.prototype.slice.call(arguments));
 
-
         // CALLBACK function
         var sendTxCallback = function(err, result) {
+            if (isCall && (method.isRevertReasonString(result) && method.abiCoder)) {
+                var reason = method.abiCoder.decodeParameter('string', '0x' + result.substring(10));
```

Comment:
Could be done with the utility functions web3.js does have but there will be the possibility in the future to have custom error signatures and the AbiCoder would then anyways be used here.